src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -541,6 +564,15 @@ public static void WriteLengthPrefixedString(BinaryWriter writer, string value)
             writer.Write(value.ToCharArray());
         }
 
+        /// <summary>
+        /// Reads the value of <see cref="CommitHashAttribute.Hash"/> of the assembly <see cref="BuildRequest"/> is defined in
+        /// </summary>
+        /// <returns>The hash value of the current assembly or an empty string</returns>
+        public static string GetCommitHash()
+        {
+            return typeof(BuildRequest).Assembly.GetCustomAttribute<CommitHashAttribute>()?.Hash;
```

Comment:
Why should we use `?.` here? Seems like that would cause us to silently omit the hash in the protocol.

Reply:
Hmm, you're right. If the hash is missing something weirder is up; it's probably fine to just crash out?